### PR TITLE
Add support for container profiles

### PR DESCRIPTION
**Pull request by shark4ce**
- Add support for providing profiles files that describes repair tools' container configuration
- Rename variables related to profiles configuration to be able to distinguish the types of profiles (e.g. repair/container profiles)
- Adapt AbstractBenchmark class to be able to pull a docker image if required

### CODE REVIEW

File: app/core/configuration.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 
 def load_configuration_details(config_file_path: str):
-    emitter.normal("\t[framework] loading profile setup")
+    emitter.normal("\t[framework] loading Fprofile setup")
```
Comment (Marti2203) [inline app/core/configuration.py]: Error?

File: app/core/container.py [hunk 2]
```diff
@@ -170,7 +173,11 @@ def get_container_id(container_name: str) -> Optional[str]:
 
 
 def build_container(
-    container_name: str, volume_list, image_name: str, cpu: str
+    container_name: str,
+    volume_list,
+    image_name: str,
+    cpu: str,
+    container_config_dict: Union[Dict[Any, Any], None] = None,
```
Comment (Marti2203) [inline app/core/container.py:180]: Can use Optional[Dict] as Optional is an alias for union with none

File: app/drivers/tools/AbstractTool.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ class AbstractTool(AbstractDriver):
     _time = stats.TimeStats()
     _space = stats.SpaceStats()
     _error = stats.ErrorStats()
-
+    current_repair_profile_id = values.current_repair_profile_id
```
Comment (Marti2203) [inline app/drivers/tools/AbstractTool.py:38]: I suggest adding this into the init

**Review by Marti2203 — COMMENTED** [review]
I will also add some commits and then we can merge it.

File: app/drivers/benchmarks/java/Vul4J.py [hunk 4]
```diff
@@ -10,8 +10,8 @@ class Vul4J(AbstractBenchmark):
 
     def __init__(self):
         self.name = os.path.basename(__file__)[:-3].lower()
-        super(Vul4J, self).__init__()
         self.image_name = "shark4ce/vul4j"
+        super(Vul4J, self).__init__()
```
[the remark this answers is not in the document]
Comment (shark4ce) [reply app/drivers/benchmarks/java/Vul4J.py]: done

File: app/drivers/tools/AbstractTool.py (hunk 3 above)
Comment (shark4ce) [reply app/drivers/tools/AbstractTool.py]: done

File: app/core/container.py (hunk 2 above)
Comment (shark4ce) [reply app/core/container.py]: done

**Review by rshariffdeen — APPROVED** [review]

**Pull request merged**